components/brave_wallet_ui/page/container.tsx

```diff
@@ -570,16 +494,102 @@ function Container (props: Props) {
         selectedButton={view}
         onSubmit={navigateTo}
       /> */}
-      <WalletSubViewLayout>
-        {renderWallet}
-        {/* {view === 'crypto' ? (
-          renderWallet
-        ) : (
-          <div style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
-            <h2>{view} view</h2>
-          </div>
-        )} */}
-      </WalletSubViewLayout>
+      <Switch>
+        <WalletSubViewLayout>
+          <Route path={WalletRoutes.Restore} exact={true}>
+            {isWalletLocked &&
+              <OnboardingRestore
+                onRestore={restoreWallet}
+                toggleShowRestore={onToggleShowRestore}
+                hasRestoreError={restorError}
+              />
+            }
+          </Route>
+          <Route path={WalletRoutes.Onboarding} exact={true}>
+            <Onboarding
+              recoveryPhrase={recoveryPhrase}
+              onPasswordProvided={passwordProvided}
+              onSubmit={completeWalletSetup}
+              onShowRestore={onToggleShowRestore}
+              // Pass a boolean if either wallet is detected
+              braveLegacyWalletDetected={false}
+              metaMaskWalletDetected={false}
+              // Pass a boolean if it failed to import
+              hasImportError={false}
+              onImportBraveLegacy={onImportBraveLegacy}
+              onImportMetaMask={onImportMetaMask}
+            />
+          </Route>
+          <Route path={WalletRoutes.Unlock} exact={true}>
+            {isWalletLocked &&
+              <LockScreen
+                onSubmit={unlockWallet}
+                disabled={inputValue === ''}
+                onPasswordChanged={handlePasswordChanged}
+                hasPasswordError={hasIncorrectPassword}
+                onShowRestore={onToggleShowRestore}
+              />
+            }
+          </Route>
+          <Route path={WalletRoutes.Backup} exact={true}>
+            <BackupWallet
+              isOnboarding={false}
+              onCancel={onHideBackup}
+              onSubmit={onBackupWallet}
+              recoveryPhrase={recoveryPhrase}
+            />
+          </Route>
+          <Route path={WalletRoutes.CryptoPage}>
+            {(isWalletCreated && !setupStillInProgress) && !isWalletLocked && walletLocation !== WalletRoutes.Backup &&
```

Comment:
Why is the `walletLocation !== WalletRoutes.Backup` required here? Similarly for `!isWalletLocked`. I thought it was handled by the other `<Route ... >` blocks, is it not?

Reply:
While the `<Route ...>` does handle what is displayed, this route is `dynamic` and not `exact`.
Inside the `Crypto` component there are additional routes for `Portfolio` and `Accounts` that share other components,
for instance the `TopNav` component. Without wrapping it in these conditions, those components will still be displayed on unwanted pages.